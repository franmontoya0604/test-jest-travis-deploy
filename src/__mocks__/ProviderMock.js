import React from 'react';

import { Router } from 'react-router-dom';

import { createBrowserHistory } from 'history';

import useInitialStateTest from "../useInitialStateTest"
import AppContext from "../context/AppContext"

const history = createBrowserHistory();





const ProviderMock = props => {
 
  const initialState = useInitialStateTest()

  return(
   <AppContext.Provider value={initialState}>
    <Router history={history}>
      {props.children}  {/*asi creamos el providerMock para simular el context en una prueba */}
    </Router>
  </AppContext.Provider>
  )};

export default ProviderMock;