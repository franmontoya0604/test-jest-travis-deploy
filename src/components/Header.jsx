import React,{useContext} from 'react'
import './styles/header.css'
import {Link} from 'react-router-dom'
import AppContext from '../context/AppContext' //asi traemos el contexto y lo indicamos en useContext



const Header = () =>{

  const {state} = useContext(AppContext);//traemos la funcion para gregar  al carrito y el estado que vamos a utilizar. cada ves que se use estos se actualiza.


    return (
        <nav className="navbar navbar-light bg-light  container-fluid">
          <h2 className="Header-title">Platzi Store</h2>
  <h2>
  <Link to='/' >  proyecto react con hooks</Link>
   </h2>
   <div className="Header-checkout">
    <Link to='/checkout'> 
     <i className="fas fa-shopping-basket "></i>
     </Link> 
     {state.cart.length > 0  && <div className="Header-alert">{state.cart.length} </div>} {/*a esto no lo ejecuta el shallow */}
     </div>
  
</nav>
    )
}

export default Header
