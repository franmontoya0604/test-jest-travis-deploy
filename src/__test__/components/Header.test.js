import React from 'react';
import { mount, shallow } from 'enzyme';
import ProviderMock from '../../__mocks__/ProviderMock';
import Header from '../../components/Header';
import renderer from 'react-test-renderer'; // de este utils usamoscreate para renderizar.



describe('<Header />', () => {
  test('Render del componente Header', () => {
    const header = shallow(
      <ProviderMock> {/*ponemos dentro del mock el componente que vaya a adquirir los datos del provider , asi aplicamos este mock*/}
        <Header />
      </ProviderMock>,
    );
    expect(header.length).toEqual(1); /*aca comprobamos que este renderizad oel header */
  });
  test('Render del Titulo', () => {
    const header = mount(
      <ProviderMock>
        <Header />
      </ProviderMock>,
    );
    expect(header.find(".Header-title").text()).toEqual('Platzi Store'); //aca vemos si el tittle de header esta bien.
  });
},

describe('header snapshot',() => {

  test('comprobar snapshot', () => {

   const header = renderer.create(
    <ProviderMock>
    <Header />
  </ProviderMock>, //aca como usamos el context para header tenemos que usar el providemock
   );
   expect(header.toJSON()).toMatchSnapshot(); 

  })


})

);
