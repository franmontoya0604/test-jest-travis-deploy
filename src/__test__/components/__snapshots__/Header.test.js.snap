// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header snapshot comprobar snapshot 1`] = `
Array [
  <nav
    className="navbar navbar-light bg-light  container-fluid"
  >
    <h2
      className="Header-title"
    >
      Platzi Store
    </h2>
    <h2>
      <a
        href="/"
        onClick={[Function]}
      >
          proyecto react con hooks
      </a>
    </h2>
    <div
      className="Header-checkout"
    >
      <a
        href="/checkout"
        onClick={[Function]}
      >
        <i
          className="fas fa-shopping-basket "
        />
      </a>
      <div
        className="Header-alert"
      >
        2
         
      </div>
       
    </div>
  </nav>,
  "  ",
]
`;
